File: solution/0000-0099/0024.Swap Nodes in Pairs/Solution.ts
function swapPairs(head: ListNode | null): ListNode | null {
    if (head == null || head.next == null) {
        return head;
    }
    const t = swapPairs(head.next.next);
    const p = head.next;
    p.next = head;
    head.next = t;
    return p;
};

File: solution/0000-0099/0024.Swap Nodes in Pairs/Solution.java
class Solution {
    public ListNode swapPairs(ListNode head) {
        if (head == null || head.next == null) {
            return head;
        }
        ListNode t = swapPairs(head.next.next);
        ListNode curr = head.next;
        curr.next = head;
        head.next = t;
        return curr;
    }
}
